### PR TITLE
installer: use unix socket

### PR DESCRIPTION
**Pull request by darkonie**
- add `docker.socket` unit. The unit will create a unix socket used by docker.

### CODE REVIEW

**Comment by darkonie** [comment]
@cmaloney @spahl this is ready for review

**Comment by spahl** [comment]
The docker package for 1.12 does not provide the systemd socket anymore. This is why this is added.

**Comment by spahl** [comment]
:ship:

**Comment by cmaloney** [comment]
Docker intentionally removed the docker.socket. that the `After=` wasn't removed as well: https://github.com/docker/docker/pull/25094

As such I'd rather not have us manually add it in. Docker 1.12 is definitely not fully baked yet.

**Comment by cmaloney** [comment]
https://github.com/docker/docker/commit/04104c3a1e6cad30cb41b762e8832215466c0e95

**Comment by spahl** [comment]
Sounds reasonable, let's move to the default socket then.

**Comment by spahl** [comment]
:ship:

**Comment by spahl** [comment]
Before we ship this we need to check if docker 1.12 does not break much more.

File: ext/dcos-installer/dcos_installer/action_lib/__init__.py [hunk 1]
```diff
@@ -340,10 +340,15 @@ def _add_prereqs_script(chain):
...
 ExecStart=
-ExecStart=/usr/bin/docker daemon --storage-driver=overlay -H fd://
+ExecStart=/usr/bin/docker daemon --storage-driver=overlay -H unix:///var/run/docker.sock
 EOF
 
-sudo yum install -y docker-engine-1.11.2
+sudo systemctl daemon-reload
+# try to stop the older docker version, but do not hard fail if it does not
+# exist.
+sudo systemctl stop docker || true
```
Comment (alberts) [inline ext/dcos-installer/dcos_installer/action_lib/__init__.py]: If we're trying to get rid of old dockers, we should probably try to stop docker.socket and docker.service?

Comment (malnick) [inline ext/dcos-installer/dcos_installer/action_lib/__init__.py]: Why true?

**Comment by cmaloney** [comment]
Hasn't had progress in a month, closing. Will try picking back up so hopefully DC/OS 1.9 is compatible with Docker 1.12

**Pull request closed without merging**